### PR TITLE
fix(progressbar): bs4 progressbar wip

### PR DESCRIPTION
**Pull request by valorkin**

### CODE REVIEW

**Comment by YevheniiaMazur** [comment]
Tested, looks good

File: demo/src/app/components/+progressbar/demos/bs4/dynamic/dynamic.html [hunk 1]
```diff
@@ -1,13 +1,13 @@
-<progress class="progress progress-striped progress-animated" [max]="max" [value]="dynamic">
+<progressbar class="progress progress-striped progress-animated" [max]="max" [value]="dynamic">
   <span style="color:white; white-space:nowrap;">{{dynamic}} / {{max}}</span>
-</progress>
+</progressbar>
 
 <small><em>No animation</em></small>
-<progress class="progress progress-success" [value]="dynamic" [max]="100"><b>{{dynamic}}%</b></progress>
+<progressbar class="progress progress-success" [value]="dynamic" [max]="100" [type]="'success'"><b>{{dynamic}}%</b></progressbar>
```
Comment (valorkin) [inline demo/src/app/components/+progressbar/demos/bs4/dynamic/dynamic.html]: string setter

File: src/modal/modal.component.ts [hunk 2]
```diff
@@ -69,8 +69,8 @@ export class ModalDirective implements AfterViewInit, OnDestroy {
   protected originalBodyPadding: number = 0;
   protected scrollbarWidth: number = 0;
 
-  protected timerHideModal: number = 0;
-  protected timerRmBackDrop: number = 0;
+  protected timerHideModal: any = 0;
```
Comment (valorkin) [inline src/modal/modal.component.ts]: merge conflict incomming

File: src/progressbar/bar.component.ts [hunk 3]
```diff
@@ -28,6 +29,10 @@ export class BarComponent implements OnInit, OnDestroy {
   public get value():number {
     return this._value;
   }
+  @HostBinding('style.width.%') get setBarWidth(){
+    this.recalculatePercentage();
+    return this.isBs3 ? 'auto' : this.percent;
```
Comment (valorkin) [inline src/progressbar/bar.component.ts]: `auto%`?

File: src/progressbar/progressbar.component.ts [hunk 4]
```diff
@@ -18,8 +27,16 @@ export class ProgressbarComponent {
   @Input() public max:number;
   /** provide one of the four supported contextual classes: `success`, `info`, `warning`, `danger` */
   @Input() public type:string;
-  /** current value of progress bar */
-  @Input() public value:number;
+  /** current value of progress bar. Could be a number or array of objects like {"value":15,"type":"info","label":"15 %"} */
+  @Input() public set value(value: number | any[]) {
+    this.stacked = Array.isArray(value);
```
Comment (valorkin) [inline src/progressbar/progressbar.component.ts]: then it should be `isStacked`

**Pull request merged**